File: .sdk-repos/orchestration-cluster-api-js/examples/tenant.ts
// Compilable usage examples for tenant management operations.
// These examples are type-checked during build to guard against API regressions.

import {
  type ClientId,
  createCamundaClient,
  type GroupId,
  type MappingRuleId,
  type RoleId,
  type TenantId,
  type Username,
} from '@camunda8/orchestration-cluster-api';

//#region CreateTenant
async function createTenantExample(tenantId: TenantId) {
  const camunda = createCamundaClient();

  const result = await camunda.createTenant({
    tenantId,
    name: 'Customer Service',
  });

  console.log(`Created tenant: ${result.tenantId}`);
}
//#endregion CreateTenant

//#region GetTenant
async function getTenantExample(tenantId: TenantId) {
  const camunda = createCamundaClient();

  const tenant = await camunda.getTenant({ tenantId }, { consistency: { waitUpToMs: 5000 } });

  console.log(`Tenant: ${tenant.name}`);
}
//#endregion GetTenant

//#region SearchTenants
async function searchTenantsExample() {
  const camunda = createCamundaClient();

  const result = await camunda.searchTenants(
    {
      page: { limit: 10 },
    },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const tenant of result.items ?? []) {
    console.log(`${tenant.tenantId}: ${tenant.name}`);
  }
}
//#endregion SearchTenants

//#region UpdateTenant
async function updateTenantExample(tenantId: TenantId) {
  const camunda = createCamundaClient();

  await camunda.updateTenant({
    tenantId,
    name: 'Customer Service Team',
  });
}
//#endregion UpdateTenant

//#region DeleteTenant
async function deleteTenantExample(tenantId: TenantId) {
  const camunda = createCamundaClient();

  await camunda.deleteTenant({ tenantId });
}
//#endregion DeleteTenant

//#region AssignUserToTenant
async function assignUserToTenantExample(tenantId: TenantId, username: Username) {
  const camunda = createCamundaClient();

  await camunda.assignUserToTenant({
    tenantId,
    username,
  });
}
//#endregion AssignUserToTenant

//#region UnassignUserFromTenant
async function unassignUserFromTenantExample(tenantId: TenantId, username: Username) {
  const camunda = createCamundaClient();

  await camunda.unassignUserFromTenant({
    tenantId,
    username,
  });
}
//#endregion UnassignUserFromTenant

//#region AssignGroupToTenant
async function assignGroupToTenantExample(tenantId: TenantId, groupId: GroupId) {
  const camunda = createCamundaClient();

  await camunda.assignGroupToTenant({
    tenantId,
    groupId,
  });
}
//#endregion AssignGroupToTenant

//#region UnassignGroupFromTenant
async function unassignGroupFromTenantExample(tenantId: TenantId, groupId: GroupId) {
  const camunda = createCamundaClient();

  await camunda.unassignGroupFromTenant({
    tenantId,
    groupId,
  });
}
//#endregion UnassignGroupFromTenant

//#region AssignRoleToTenant
async function assignRoleToTenantExample(tenantId: TenantId, roleId: RoleId) {
  const camunda = createCamundaClient();

  await camunda.assignRoleToTenant({
    tenantId,
    roleId,
  });
}
//#endregion AssignRoleToTenant

//#region UnassignRoleFromTenant
async function unassignRoleFromTenantExample(tenantId: TenantId, roleId: RoleId) {
  const camunda = createCamundaClient();

  await camunda.unassignRoleFromTenant({
    tenantId,
    roleId,
  });
}
//#endregion UnassignRoleFromTenant

//#region AssignClientToTenant
async function assignClientToTenantExample(tenantId: TenantId, clientId: ClientId) {
  const camunda = createCamundaClient();

  await camunda.assignClientToTenant({
    tenantId,
    clientId,
  });
}
//#endregion AssignClientToTenant

//#region UnassignClientFromTenant
async function unassignClientFromTenantExample(tenantId: TenantId, clientId: ClientId) {
  const camunda = createCamundaClient();

  await camunda.unassignClientFromTenant({
    tenantId,
    clientId,
  });
}
//#endregion UnassignClientFromTenant

//#region AssignMappingRuleToTenant
async function assignMappingRuleToTenantExample(tenantId: TenantId, mappingRuleId: MappingRuleId) {
  const camunda = createCamundaClient();

  await camunda.assignMappingRuleToTenant({
    tenantId,
    mappingRuleId,
  });
}
//#endregion AssignMappingRuleToTenant

//#region UnassignMappingRuleFromTenant
async function unassignMappingRuleFromTenantExample(
  tenantId: TenantId,
  mappingRuleId: MappingRuleId
) {
  const camunda = createCamundaClient();

  await camunda.unassignMappingRuleFromTenant({
    tenantId,
    mappingRuleId,
  });
}
//#endregion UnassignMappingRuleFromTenant

//#region SearchUsersForTenant
async function searchUsersForTenantExample(tenantId: TenantId) {
  const camunda = createCamundaClient();

  const result = await camunda.searchUsersForTenant(
    { tenantId },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const user of result.items ?? []) {
    console.log(`Tenant member: ${user.username}`);
  }
}
//#endregion SearchUsersForTenant

//#region SearchClientsForTenant
async function searchClientsForTenantExample(tenantId: TenantId) {
  const camunda = createCamundaClient();

  const result = await camunda.searchClientsForTenant(
    { tenantId },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const client of result.items ?? []) {
    console.log(`Client: ${client.clientId}`);
  }
}
//#endregion SearchClientsForTenant

//#region SearchGroupIdsForTenant
async function searchGroupIdsForTenantExample(tenantId: TenantId) {
  const camunda = createCamundaClient();

  const result = await camunda.searchGroupIdsForTenant(
    { tenantId },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const group of result.items ?? []) {
    console.log(`Group: ${group.groupId}`);
  }
}
//#endregion SearchGroupIdsForTenant

//#region SearchRolesForTenant
async function searchRolesForTenantExample(tenantId: TenantId) {
  const camunda = createCamundaClient();

  const result = await camunda.searchRolesForTenant(
    { tenantId },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const role of result.items ?? []) {
    console.log(`Role: ${role.name}`);
  }
}
//#endregion SearchRolesForTenant

//#region SearchMappingRulesForTenant
async function searchMappingRulesForTenantExample(tenantId: TenantId) {
  const camunda = createCamundaClient();

  const result = await camunda.searchMappingRulesForTenant(
    { tenantId },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const rule of result.items ?? []) {
    console.log(`Mapping rule: ${rule.name}`);
  }
}
//#endregion SearchMappingRulesForTenant

// Suppress "declared but never read"
void createTenantExample;
void getTenantExample;
void searchTenantsExample;
void updateTenantExample;
void deleteTenantExample;
void assignUserToTenantExample;
void unassignUserFromTenantExample;
void assignGroupToTenantExample;
void unassignGroupFromTenantExample;
void assignRoleToTenantExample;
void unassignRoleFromTenantExample;
void assignClientToTenantExample;
void unassignClientFromTenantExample;
void assignMappingRuleToTenantExample;
void unassignMappingRuleFromTenantExample;
void searchUsersForTenantExample;
void searchClientsForTenantExample;
void searchGroupIdsForTenantExample;
void searchRolesForTenantExample;
void searchMappingRulesForTenantExample;

// Suppress "declared but never read"
void createTenantExample;
void getTenantExample;
void searchTenantsExample;
void updateTenantExample;
void deleteTenantExample;
void assignUserToTenantExample;
void unassignUserFromTenantExample;
void assignGroupToTenantExample;
void unassignGroupFromTenantExample;
void assignRoleToTenantExample;
void unassignRoleFromTenantExample;
void assignClientToTenantExample;
void unassignClientFromTenantExample;
void assignMappingRuleToTenantExample;
void unassignMappingRuleFromTenantExample;
void searchUsersForTenantExample;
void searchClientsForTenantExample;
void searchGroupIdsForTenantExample;
void searchRolesForTenantExample;
void searchMappingRulesForTenantExample;

File: .sdk-repos/orchestration-cluster-api-csharp/examples/Tenant.cs
// Compilable usage examples for tenant management operations.
// These examples are type-checked during build to guard against API regressions.
using Camunda.Orchestration.Sdk;

public static class TenantExamples
{
    #region CreateTenant
    // <CreateTenant>
    public static async Task CreateTenantExample(TenantId tenantId)
    {
        using var client = CamundaClient.Create();

        var result = await client.CreateTenantAsync(new TenantCreateRequest
        {
            TenantId = tenantId,
            Name = "Acme Corporation",
        });

        Console.WriteLine($"Tenant key: {result.TenantId}");
    }
    // </CreateTenant>
    #endregion CreateTenant

    #region GetTenant

    // <GetTenant>
    public static async Task GetTenantExample(TenantId tenantId)
    {
        using var client = CamundaClient.Create();

        var result = await client.GetTenantAsync(tenantId);
        Console.WriteLine($"Tenant: {result.Name}");
    }
    // </GetTenant>
    #endregion GetTenant

    #region SearchTenants

    // <SearchTenants>
    public static async Task SearchTenantsExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchTenantsAsync(new TenantSearchQueryRequest());

        foreach (var tenant in result.Items)
        {
            Console.WriteLine($"Tenant: {tenant.Name}");
        }
    }
    // </SearchTenants>
    #endregion SearchTenants

    #region UpdateTenant

    // <UpdateTenant>
    public static async Task UpdateTenantExample(TenantId tenantId)
    {
        using var client = CamundaClient.Create();

        await client.UpdateTenantAsync(
            tenantId,
            new TenantUpdateRequest
            {
                Name = "Acme Corp International",
            });
    }
    // </UpdateTenant>
    #endregion UpdateTenant

    #region DeleteTenant

    // <DeleteTenant>
    public static async Task DeleteTenantExample(TenantId tenantId)
    {
        using var client = CamundaClient.Create();

        await client.DeleteTenantAsync(tenantId);
    }
    // </DeleteTenant>
    #endregion DeleteTenant

    #region AssignUserToTenant

    // <AssignUserToTenant>
    public static async Task AssignUserToTenantExample(TenantId tenantId, Username username)
    {
        using var client = CamundaClient.Create();

        await client.AssignUserToTenantAsync(
            tenantId,
            username);
    }
    // </AssignUserToTenant>
    #endregion AssignUserToTenant

    #region UnassignUserFromTenant

    // <UnassignUserFromTenant>
    public static async Task UnassignUserFromTenantExample(TenantId tenantId, Username username)
    {
        using var client = CamundaClient.Create();

        await client.UnassignUserFromTenantAsync(
            tenantId,
            username);
    }
    // </UnassignUserFromTenant>
    #endregion UnassignUserFromTenant

    #region AssignGroupToTenant

    // <AssignGroupToTenant>
    public static async Task AssignGroupToTenantExample(TenantId tenantId, GroupId groupId)
    {
        using var client = CamundaClient.Create();

        await client.AssignGroupToTenantAsync(
            tenantId,
            groupId);
    }
    // </AssignGroupToTenant>
    #endregion AssignGroupToTenant

    #region UnassignGroupFromTenant

    // <UnassignGroupFromTenant>
    public static async Task UnassignGroupFromTenantExample(TenantId tenantId, GroupId groupId)
    {
        using var client = CamundaClient.Create();

        await client.UnassignGroupFromTenantAsync(
            tenantId,
            groupId);
    }
    // </UnassignGroupFromTenant>
    #endregion UnassignGroupFromTenant

    #region AssignRoleToTenant

    // <AssignRoleToTenant>
    public static async Task AssignRoleToTenantExample(TenantId tenantId, RoleId roleId)
    {
        using var client = CamundaClient.Create();

        await client.AssignRoleToTenantAsync(
            tenantId,
            roleId);
    }
    // </AssignRoleToTenant>
    #endregion AssignRoleToTenant

    #region UnassignRoleFromTenant

    // <UnassignRoleFromTenant>
    public static async Task UnassignRoleFromTenantExample(TenantId tenantId, RoleId roleId)
    {
        using var client = CamundaClient.Create();

        await client.UnassignRoleFromTenantAsync(
            tenantId,
            roleId);
    }
    // </UnassignRoleFromTenant>
    #endregion UnassignRoleFromTenant

    #region AssignClientToTenant

    // <AssignClientToTenant>
    public static async Task AssignClientToTenantExample(TenantId tenantId, ClientId clientId)
    {
        using var client = CamundaClient.Create();

        await client.AssignClientToTenantAsync(
            tenantId,
            clientId);
    }
    // </AssignClientToTenant>
    #endregion AssignClientToTenant

    #region UnassignClientFromTenant

    // <UnassignClientFromTenant>
    public static async Task UnassignClientFromTenantExample(TenantId tenantId, ClientId clientId)
    {
        using var client = CamundaClient.Create();

        await client.UnassignClientFromTenantAsync(
            tenantId,
            clientId);
    }
    // </UnassignClientFromTenant>
    #endregion UnassignClientFromTenant

    #region AssignMappingRuleToTenant

    // <AssignMappingRuleToTenant>
    public static async Task AssignMappingRuleToTenantExample(TenantId tenantId, MappingRuleId mappingRuleId)
    {
        using var client = CamundaClient.Create();

        await client.AssignMappingRuleToTenantAsync(
            tenantId,
            mappingRuleId);
    }
    // </AssignMappingRuleToTenant>
    #endregion AssignMappingRuleToTenant

    #region UnassignMappingRuleFromTenant

    // <UnassignMappingRuleFromTenant>
    public static async Task UnassignMappingRuleFromTenantExample(TenantId tenantId, MappingRuleId mappingRuleId)
    {
        using var client = CamundaClient.Create();

        await client.UnassignMappingRuleFromTenantAsync(
            tenantId,
            mappingRuleId);
    }
    // </UnassignMappingRuleFromTenant>
    #endregion UnassignMappingRuleFromTenant

    #region SearchUsersForTenant

    // <SearchUsersForTenant>
    public static async Task SearchUsersForTenantExample(TenantId tenantId)
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchUsersForTenantAsync(
            tenantId,
            new TenantUserSearchQueryRequest());

        foreach (var user in result.Items)
        {
            Console.WriteLine($"User: {user.Username}");
        }
    }
    // </SearchUsersForTenant>
    #endregion SearchUsersForTenant

    #region SearchClientsForTenant

    // <SearchClientsForTenant>
    public static async Task SearchClientsForTenantExample(TenantId tenantId)
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchClientsForTenantAsync(
            tenantId,
            new TenantClientSearchQueryRequest());

        foreach (var c in result.Items)
        {
            Console.WriteLine($"Client: {c.ClientId}");
        }
    }
    // </SearchClientsForTenant>
    #endregion SearchClientsForTenant

    #region SearchGroupIdsForTenant

    // <SearchGroupIdsForTenant>
    public static async Task SearchGroupIdsForTenantExample(TenantId tenantId)
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchGroupIdsForTenantAsync(
            tenantId,
            new TenantGroupSearchQueryRequest());

        foreach (var group in result.Items)
        {
            Console.WriteLine($"Group: {group.GroupId}");
        }
    }
    // </SearchGroupIdsForTenant>
    #endregion SearchGroupIdsForTenant

    #region SearchRolesForTenant

    // <SearchRolesForTenant>
    public static async Task SearchRolesForTenantExample(TenantId tenantId)
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchRolesForTenantAsync(
            tenantId,
            new RoleSearchQueryRequest());

        foreach (var role in result.Items)
        {
            Console.WriteLine($"Role: {role.Name}");
        }
    }
    // </SearchRolesForTenant>
    #endregion SearchRolesForTenant

    #region SearchMappingRulesForTenant

    // <SearchMappingRulesForTenant>
    public static async Task SearchMappingRulesForTenantExample(TenantId tenantId)
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchMappingRulesForTenantAsync(
            tenantId,
            new MappingRuleSearchQueryRequest());

        foreach (var rule in result.Items)
        {
            Console.WriteLine($"Mapping rule: {rule.MappingRuleId}");
        }
    }
    // </SearchMappingRulesForTenant>
    #endregion SearchMappingRulesForTenant
}
